File: Batch3_VishnuNair_ShopEZ– E-Commerce Application/ShopEZ/services/OrderService/Models/Order.cs
namespace OrderService.Models
{
    public class Order
    {
        public int OrderId { get; set; }
        public int UserId { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public decimal TotalAmount { get; set; }
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
    public class OrderItem
    {
        public int OrderItemId { get; set; }
        public int OrderId { get; set; }
        public int ProductId { get; set; }
        public string ProductName { get; set; } = string.Empty;
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public Order? Order { get; set; }
    }
}

File: Batch3_VishnuNair_ShopEZ– E-Commerce Application/ecommerce-angular fullstack/src/app/models/order.ts
export interface CartItem {
  productId: number;
  quantity: number;
}

export interface CreateOrder {
  userId: number;
  cartItems: CartItem[];
}

export interface OrderItem {
  orderItemId: number;
  productId: number;
  productName: string;
  quantity: number;
  price: number;
}

export interface Order {
  orderId: number;
  userId: number;
  orderDate: string;
  totalAmount: number;
  items: OrderItem[];
}
